aioauth/grant_type.py

```diff
@@ -123,6 +143,18 @@ async def validate_request(self, request: Request) -> Client:
 
 
 class PasswordGrantType(GrantTypeBase):
+    """
+    The Password grant type is a way to exchange a user's credentials
```

Comment:
should we remove the `PasswordGrantType` from the `BaseAuthorizationServer` by default?

Reply:
I would advise against removing it. Password grant, while not recommended for _every_ client, is the preferred method for those that separate their backend and frontend. In other words, if you have a FastAPI OAuth server running and a React frontend, the only realistic way to have the user login is via the `PasswordGrantType` where the React frontend sends the relevant info to the server (this is also common for mobile apps). See [here](https://developer.okta.com/blog/2018/06/29/what-is-the-oauth2-password-grant#when-to-use-the-password-grant-type) for more info.

While I haven't gotten to checking this just yet, one thing that authorization servers *must* have is the ability to specify _which_ grant type a specific client has access to. Company A might have a OAuth server and frontend and mobile app that leverages the `PasswordGrantType`, while also allowing company B to use their OAuth server, but only have access to the `AuthorizationCodeGrantType`.

Reply:
no, I did not mean complete deletion, as this is part of the oauth2 protocol. I thought about excluding it by default from BaseAuthorizationServer: 

https://github.com/aliev/aioauth/blob/188699eaeb1c1c7d1e4223ccd64641dcfc7037f9/src/aioauth/base/server.py#L25-L28

user can still enable it with the registry method:

https://github.com/aliev/aioauth/blob/188699eaeb1c1c7d1e4223ccd64641dcfc7037f9/src/aioauth/base/server.py#L34-L41

Reply:
Oh, I see! 

I'm in favor, in such case. I haven't gotten to documenting the base server just yet, so I didn't know that function even existed. We can document this properly and ensure users are aware that there is the _option_, even though it's not recommended. Whatever the case, I wouldn't be making the modifications via this PR since it's exclusive to documentation, but I can add to the docs as soon as we make that change.

Reply:
Indeed. Better to just document it and user will decide should it remove it or keep it